woocommerce-gateway-ebanx/gateways/class-wc-ebanx-credit-card-gateway.php

```diff
@@ -255,34 +255,39 @@ protected function save_user_meta_fields($order)
 	{
 		parent::save_user_meta_fields($order);
 
-		if ($this->userId && $this->configs->settings['save_card_data'] === 'yes' && isset($_POST['ebanx-save-credit-card']) && $_POST['ebanx-save-credit-card'] === 'yes') {
-			$cards = get_user_meta($this->userId, '_ebanx_credit_card_token', true);
-			$cards = !empty($cards) ? $cards : [];
+		if ($this->userId
+			|| $this->get_setting_or_default('save_card_data', 'no') !== 'yes'
+			|| ! WC_EBANX_Request::has(['ebanx-save-credit-card'])
+			|| WC_EBANX_Request::read('ebanx-save-credit-card') !== 'yes') {
+			return;
+		}
 
-			$card = new \stdClass();
+		$cards = get_user_meta($this->userId, '_ebanx_credit_card_token', true);
+		$cards = !empty($cards) ? $cards : [];
 
-			$card->brand = $_POST['ebanx_brand'];
-			$card->token = $_POST['ebanx_token'];
-			$card->masked_number = $_POST['ebanx_masked_card_number'];
+		$card = new \stdClass();
 
-			foreach ($cards as $cd) {
-				if (empty($cd)) {
-					continue;
-				}
+		$card->brand = $_POST['ebanx_brand'];
+		$card->token = $_POST['ebanx_token'];
+		$card->masked_number = $_POST['ebanx_masked_card_number'];
```

Comment:
Instead of using the $_POST super global it's preferred that we use `WC_EBANX_Request::read($key);` in these cases.